Makefile

```diff
@@ -168,6 +168,22 @@ dep-ensure: ##@dependencies Dep ensure and apply all patches
 dep-install: ##@dependencies Install vendoring tool
 	go get -u github.com/golang/dep/cmd/dep
 
+update-geth: ##@dependencies Update geth (use GETH_BRANCH to optionally set the geth branch name)
+ifdef GETH_BRANCH
+	@# escape slashes
+	@GETH_BRANCH=$(echo $GETH_BRANCH | sed 's@\/@\\\/@g')
+	@# Update go-ethereum contraint branch
+	@sed -i 'N;N;s@\(\[\[constraint]]\n  name = "github.com\/ethereum\/go-ethereum"\n  branch =\)\(.*\)@\1 '"\"${GETH_BRANCH}\""'@g' Gopkg.toml
+endif
+	@dep ensure -v -update github.com/ethereum/go-ethereum
+	$(MAKE) dep-ensure
+	git add Gopkg.lock Gopkg.toml vendor/ _assets/patches/
+ifeq ($(git diff --cached --quiet), 0)
+	@echo "No changes to commit. Geth up to date." && exit 1
```

Comment:
Not really sure if we should exit with `1` if there was nothing to update. It seems like a valid scenario.

Reply:
Agreed. I actually wasn't sure how else to make the Makefile exit early. This will be fixed once this is pulled into a bash script :+1: Thanks for the comments!